src/adapter/src/catalog/apply.rs

```diff
@@ -1304,8 +1336,16 @@ impl CatalogState {
             let (handle, _idx, remaining) = future::select_all(handles).await;
             handles = remaining;
             let (id, res) = handle.expect("must join");
+            let mut insert_cached_expr = |cached_expr| {
+                if let Some(cached_expr) = cached_expr {
+                    local_expression_cache.insert_cached_expression(id, cached_expr);
+                }
+            };
             match res {
-                Ok(item) => {
+                Ok((item, uncached_expr)) => {
+                    if let Some((uncached_expr, optimizer_features)) = uncached_expr {
+                        local_expression_cache.insert_uncached_expression(id, uncached_expr, optimizer_features);
```

Comment:
What's this business of cached and uncached expressions? Might be easier to quickly huddle once you're online.

Aha! Now that I read further, it's: uncached expressions = expressions that were not cached but _should_ be added to the cache once we update it. When I first read this, I thought these might be expressions that we don't want to cache or something... :see_no_evil:

Reply:
Yes that's correct. This was one of the most painful parts of this PR, plumbing everything around to various parts of startup. I've added a doc-comment to this method to try and make it clearer.